accounts/abi/bind/backends/blockchain.go

```diff
@@ -123,18 +124,17 @@ func (b *BlockchainContractCaller) callContract(call klaytn.CallMsg, block *type
 }
 
 func (b *BlockchainContractCaller) getBlockAndState(num *big.Int) (*types.Block, *state.StateDB, error) {
-	var header *types.Header
+	var block *types.Block
 	if num == nil {
-		header = b.bc.CurrentHeader()
+		block = b.bc.CurrentBlock()
 	} else {
-		header = b.bc.GetHeaderByNumber(num.Uint64())
+		header := b.bc.GetHeaderByNumber(num.Uint64())
+		if header == nil {
+			return nil, nil, errBlockDoesNotExist
+		}
+		block = b.bc.GetBlock(header.Hash(), header.Number.Uint64())
```

Comment:
`block` can be `nil`, then it will cause panic in the following code by calling `block.Root()`.

Reply:
@aidan-kwon Good point, thanks! I'll return `errBlockNotFound` for both cases - 1) header not found, and 2) header found but corresponding block not found.